core/src/main/kotlin/com/epam/drill/admindata/BuildManager.kt

```diff
@@ -61,14 +55,21 @@ class AgentBuildManager(val agentId: String, val storeClient: StoreClient, lastB
         )
     }
 
-    suspend fun compareToPrev(buildVersion: String) {
+    suspend fun processBuild(
+        notificationsManager: NotificationsManager,
+        agentInfo: AgentInfo
+    ) {
+        val buildVersion = agentInfo.buildVersion
         val currentMethods = buildInfos[buildVersion]?.classesBytes?.mapValues { (className, bytes) ->
             BcelClassParser(bytes, className).parseToJavaMethods()
         } ?: emptyMap()
         buildInfos[buildVersion] = buildInfos[buildVersion]?.copy(javaMethods = currentMethods)
             ?: BuildInfo(buildVersion = buildVersion)
         val buildInfo = buildInfos[buildVersion] ?: BuildInfo(buildVersion = buildVersion)
-        val prevMethods = buildInfos[buildInfo.prevBuild]?.javaMethods ?: mapOf()
+
+        val buildVersionIsNew = buildInfos[buildVersion] == null || buildInfos[buildVersion]!!.new
```

Comment:
what is that thing???

Reply:
done to check if this build version have already been processed in the past or not

Reply:
srsly?!!!!